apiserver/facades/agent/storageprovisioner/storageprovisioner.go

```diff
@@ -1015,13 +1019,14 @@ func (s *StorageProvisionerAPIv4) VolumeAttachmentParams(
 		}
 		// Volumes can be attached to units (caas models) or machines.
 		// We only care about instance id for machine attachments.
-		var instanceId instance.Id
+		var instanceId string
```

Comment:
Why do we return a string for an instance.Id? I think we're losing type correctness here.

Reply:
I think it was done to keep consistency with the UUIDs which were also strings on machine. Do you think I should set it back to `instance.Id()`? I can do that in a following PR.

Reply:
I agree with Simon. Keep core (or domain) instance id type. That the pattern we're following elsewhere

Reply:
Just to be clear, in this particular method, casting to an instance.Id is useless because the only place it's used is on the `InstanceID` field of the returned `VolumeAttachmentParams` which is a `string`. 
If what you would like is that the service method returns an instance.Id altogether then I can do that in a follow up PR

Reply:
Do in a follow-up PR.